master/internal/api_workspace.go

```diff
@@ -579,19 +591,23 @@ func (a *apiServer) DeleteWorkspace(
 	log.Debugf("deleting workspace %d templates", req.Id)
 	err = templates.DeleteWorkspaceTemplates(ctx, int(req.Id))
 	if err != nil {
-		return nil, errors.Wrapf(err, "error deleting workspace (%d) templates", req.Id)
+		return nil, fmt.Errorf("error deleting workspace (%d) templates", req.Id)
 	}
 
 	if len(projects) == 0 {
 		err = a.m.db.QueryProto("delete_workspace", holder, req.Id)
-		return &apiv1.DeleteWorkspaceResponse{Completed: (err == nil)},
-			errors.Wrapf(err, "error deleting workspace (%d)", req.Id)
+		if err != nil {
+			return nil, fmt.Errorf("error deleting workspace (%d)", req.Id)
+		}
+		return &apiv1.DeleteWorkspaceResponse{Completed: true}, nil
 	}
 	go func() {
 		a.deleteWorkspace(ctx, req.Id, projects)
 	}()
-	return &apiv1.DeleteWorkspaceResponse{Completed: false},
-		errors.Wrapf(err, "error deleting workspace (%d)", req.Id)
+	if err != nil {
+		return nil, fmt.Errorf("error deleting workspace (%d)", req.Id)
+	}
+	return &apiv1.DeleteWorkspaceResponse{Completed: true}, nil
```

Comment:
```suggestion
	return &apiv1.DeleteWorkspaceResponse{Completed: false}, nil
```

Reply:
I don't understand; isn't the request completed? Earlier, if err == nil we return `Completed: true` right?
from original line 587:
```
if len(projects) == 0 {
		err = a.m.db.QueryProto("delete_workspace", holder, req.Id)
		return &apiv1.DeleteWorkspaceResponse{Completed: (err == nil)},
			errors.Wrapf(err, "error deleting workspace (%d)", req.Id)
```

Reply:
No at this point the request is not completed, notice how we call ```deleteWorkspaces``` in a background goroutine when len(projects) > 0.

```
go func() {
    a.deleteWorkspace(ctx, req.Id, projects)
}()
```

Original line 587 maps to new 602. This is for original line 593.

Reply:
I understand now; thank you!